packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts

```diff
@@ -55,44 +54,70 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
     });
 
     execution.processExitSubject.subscribe(async exitCode => {
-      const telemetry = new TelemetryBuilder();
-      let success = false;
-      try {
-        BaseDeployExecutor.errorCollection.clear();
-        if (stdOut) {
-          const deployParser = new ForceDeployResultParser(stdOut);
-          const errors = deployParser.getErrors();
-          if (errors && !deployParser.hasConflicts()) {
-            channelService.showChannelOutput();
+      await this.exitProcessHandlerDeploy(
+        exitCode,
+        stdOut,
+        workspacePath,
+        execFilePathOrPaths,
+        execution,
+        startTime,
+        cancellationToken,
+        cancellationTokenSource
+      );
+    });
+  }
+
+  protected async exitProcessHandlerDeploy(
+    exitCode: number | undefined,
+    stdOut: string,
+    workspacePath: string,
+    execFilePathOrPaths: string,
+    execution: CliCommandExecution,
+    startTime: [number, number],
+    cancellationToken: vscode.CancellationToken | undefined,
+    cancellationTokenSource: vscode.CancellationTokenSource
+  ): Promise<void> {
+    if (exitCode === 0 && this.getDeployType() === DeployType.Push) {
+      const pushResult = JSON.parse(stdOut);
```

Comment:
same q here about the JSON.parse.  what happen if the parse throws?

Reply:
Here it outputs to the log and ends with a 0 exit code
<img width="600" alt="2023-05-19_16-01-33" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/46458081/1f8bd21f-e448-4654-8632-268ed7889d2e">